digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadAddressValidator.scala

```diff
@@ -0,0 +1,31 @@
+package fi.liikennevirasto.viite
+
+import fi.liikennevirasto.viite.dao.{ProjectDAO, RoadAddressDAO, RoadAddressProject}
+import org.joda.time.format.DateTimeFormat
+
+object RoadAddressValidator {
+
+  def checkAvailable(number: Long, part: Long, currentProject: RoadAddressProject) = {
+    val isReserved = RoadAddressDAO.isNotAvailableForProject(number, part, currentProject.id)
+    if (!isReserved) {
+      None
+    } else {
+      val fmt = DateTimeFormat.forPattern("dd.MM.yyyy")
+      throw new ProjectValidationException(
+        s"TIE $number OSA $part on jo olemassa projektin alkupäivänä ${currentProject.startDate.toString(fmt)}, tarkista tiedot")
+    }
+  }
+
+  def checkNotReserved(number: Long, part: Long, currentProject: RoadAddressProject) = {
+    val project = ProjectDAO.roadPartReservedByProject(number, part, currentProject.id, withProjectId = true)
+    if (project.nonEmpty) {
+      throw new ProjectValidationException(s"TIE $number OSA $part on jo varattuna projektissa ${project.get}, tarkista tiedot")
+    }
+  }
+
+  def fetchProject(id: Long): RoadAddressProject = {
```

Comment:
In my opinion this shouldn't be part of validator, but if changed to checkProjectExists(id: Long) and being Unit then yes.